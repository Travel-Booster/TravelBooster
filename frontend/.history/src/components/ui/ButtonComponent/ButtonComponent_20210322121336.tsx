import React, { memo } from 'react'
import { StyleSheet, Button } from 'react-native'

/**
 * Props types
 */
type ButtonComponentProps = {
    children: any,
    // onClick: Function,
    // variant: string,
    // size: string,
    // width: string
}

/**
 * Button Component
 * @description Uses message function to return desired text
 * @param {any} children - button content
 * @param {function} onClick - button action
 * @returns {object} - return component with children
 */
const ButtonComponent = memo<ButtonComponentProps>(({ 
    children, 
    // onClick, 
    // variant,
    // size,
    // width 
}) => {
    return (
        <Button
        title={children}
            // onClick={onClick()}
            style={styles.button} />

    )
})

const styles = StyleSheet.create({
    button: {
        marginTop: 50,
        // backgroundColor: 'red'
    },
})

export default ButtonComponent
