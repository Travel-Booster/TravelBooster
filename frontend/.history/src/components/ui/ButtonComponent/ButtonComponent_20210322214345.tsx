import React, { memo } from 'react'
import { StyleSheet, Button } from 'react-native'

/**
 * Props types
 */
type ButtonComponentProps = {
    children: any,
    // onClick: Function,
    // variant: string,
    // size: string,
    // width: string
}

/**
 * Button Component
 * @description Uses message function to return desired text
 * @param {any} children - button content
 * @param {function} onClick - button action
 * @returns {object} - return component with children
 */
const ButtonComponent = memo<ButtonComponentProps>(({ 
    children, 
    // onClick, 
    // variant,
    // size,
    // width 
}) => {
    return (
        <Button
  onPress={onPressLearnMore}
  title="Learn More"
  color="#841584"
  accessibilityLabel="Learn more about this purple button"
/>
    )
})

const styles = StyleSheet.create({
    button: {
        marginTop: 50,
        // backgroundColor: 'red'
    },
})

export default ButtonComponent
