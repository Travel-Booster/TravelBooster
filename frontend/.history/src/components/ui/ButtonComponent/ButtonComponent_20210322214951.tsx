import React, { memo } from 'react'
import { StyleSheet, Button, View } from 'react-native'

/**
 * Props types
 */
type ButtonComponentProps = {
    children: any
}

/**
 * Button Component
 * @description Uses message function to return desired text
 * @param {any} children - button content
 * @param {function} onClick - button action
 * @returns {object} - return component with children
 */
const ButtonComponent = memo<ButtonComponentProps>(({ 
    children
}) => {
    return (
        <View style={styles.button}>
            <Button
                onPress={() => console.log('dupa')}
                title={children}
                color="#841584"
                accessibilityLabel="Learn more about this purple button" />
        </View>
    )
})

const styles = StyleSheet.create({
    button: {
        marginTop: 50,
        padding: '20, 80'
        // backgroundColor: 'red'
    },
})

export default ButtonComponent
