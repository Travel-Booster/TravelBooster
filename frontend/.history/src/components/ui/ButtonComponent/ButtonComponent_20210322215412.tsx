import React, { memo } from 'react'
import { StyleSheet, TouchableOpacity, View } from 'react-native'

/**
 * Props types
 */
type ButtonComponentProps = {
    children: any
}

/**
 * Button Component
 * @description Uses message function to return desired text
 * @param {any} children - button content
 * @param {function} onClick - button action
 * @returns {object} - return component with children
 */
const ButtonComponent = memo<ButtonComponentProps>(({ 
    children
}) => {
    return (
        <View style={styles.button}>
            <TouchableOpacity
                onPress={() => console.log('dupa')}>
                {children}
            </TouchableOpacity>
        </View>
    )
})

const styles = StyleSheet.create({
    button: {
        marginTop: 50,
        padding: '20px 80px',
        backgroundColor: 'red'
    },
})

export default ButtonComponent
