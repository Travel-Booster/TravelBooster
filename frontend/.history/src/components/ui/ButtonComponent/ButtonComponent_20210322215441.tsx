import React, { memo } from 'react'
import { StyleSheet, TouchableOpacity, View } from 'react-native'

/**
 * Props types
 */
type ButtonComponentProps = {
    children: any
}

/**
 * Button Component
 * @description Uses message function to return desired text
 * @param {any} children - button content
 * @param {function} onClick - button action
 * @returns {object} - return component with children
 */
const ButtonComponent = memo<ButtonComponentProps>(({ 
    children
}) => {
    return (
        <TouchableOpacity
            onPress={() => console.log('dupa')}
            style={styles.button} >
            {children}
        </TouchableOpacity>
    )
})

const styles = StyleSheet.create({
    button: {
        marginTop: 50,
        padding: '20px 80px',
        backgroundColor: 'red'
    },
})

export default ButtonComponent
