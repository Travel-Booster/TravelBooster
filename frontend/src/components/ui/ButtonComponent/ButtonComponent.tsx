import React, { memo } from 'react'
import { StyleSheet, TouchableOpacity } from 'react-native'
import colors from 'services/styles/colors'
import TextComponent from 'components/ui/TextComponent'

/**
 * Props types
 */
type ButtonComponentProps = {
    text: string,
    action: () => void,
    color?: string,
    width?: string
}

/**
 * Button Component
 * @component
 * @description Button Component
 * @param {any} text - button content
 * @param {function} action - button action
 * @param {string} color - button color, default: 'primary'
 * @param {string} width - button width, default: 'none'
 * @returns {object} - return component with children
 */
const ButtonComponent = memo<ButtonComponentProps>(({
    text,
    action,
    color,
    width
}) => {
    return (
        <TouchableOpacity
            onPress={action}
            style={{
                ...styles.button,
                backgroundColor: color === 'error'
                    ? colors.LIGHT_BUTTON_ERROR
                    : colors.LIGHT_BUTTON_PRIMARY,
                width: width === 'full'
                    ? '100%'
                    : 'auto'
            }}>
            <TextComponent 
                text={text}
                variant='button' />
        </TouchableOpacity>
    )
})

const styles = StyleSheet.create({
    button: {
        paddingHorizontal: 20,
        paddingVertical: 8,
        borderRadius: 20, 
        alignContent: 'center',
        alignItems: 'center'
    }
})

export default ButtonComponent
