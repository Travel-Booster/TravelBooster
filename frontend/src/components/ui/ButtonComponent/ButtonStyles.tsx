import { StyleSheet } from 'react-native'
import colors from 'services/styles/colors'

export default StyleSheet.create({
  button: {
    width: 'auto',
    paddingHorizontal: 20,
    paddingVertical: 8,
    borderRadius: 20, 
    alignContent: 'center',
    alignItems: 'center'
  },
  primary: {
    backgroundColor: colors.LIGHT_BUTTON_PRIMARY
  },
  error: {
    backgroundColor: colors.LIGHT_BUTTON_ERROR
  },
  full: {
    width: '100%'
  }
})
