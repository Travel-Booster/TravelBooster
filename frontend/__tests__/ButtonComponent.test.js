import React from 'react'
import renderer from 'react-test-renderer'
import ButtonComponent from 'components/ui/ButtonComponent'

const props = {
    text: 'hello',
    action: () => console.log('test'),
    color: 'error',
    width: 'full'
}

const component = <ButtonComponent {...props} />
let wrapper = renderer.create(component).toJSON()

it('<TextComponent /> renders', () => {
    expect(wrapper).toBeTruthy()
})
    
it('<TextComponent /> test snapshot', () => {
    expect(wrapper).toMatchSnapshot()
})
