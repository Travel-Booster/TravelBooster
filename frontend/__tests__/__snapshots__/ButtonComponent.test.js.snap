// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextComponent /> test snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignContent": "center",
      "alignItems": "center",
      "backgroundColor": "#FF736A",
      "borderRadius": 20,
      "opacity": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 8,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontFamily": "montserrat-semi-bold",
        "fontSize": 14,
      }
    }
  >
    Click me!
  </Text>
</View>
`;
